### DISCUSSION TITLE
Error on page transitions when using with Turbo

### DISCUSSION THREAD

**Discussion by tonysm** [alpinejs/alpine#3561] [7. Bugs (archived)]
Hey, folks.

I'm getting this error on the console:

<details>
<summary>Screenshot</summary>

<img width="1159" alt="Screenshot 2023-05-09 at 21 42 25" src="https://github.com/alpinejs/alpine/assets/1178621/b514f1fd-cc8a-409a-917d-d11edfee7c34">

</details>

It looks like it's related to `[x-transition]` (looked for `isFromCancelledTransition` in the code base, found it [here](https://github.com/alpinejs/alpine/blob/61bef0dcf151202c0fa3553465dd86cfb991046c/packages/alpinejs/src/directives/x-transition.js#L185))

I was able to pinpoint it to version `3.10.3` ([release notes](https://github.com/alpinejs/alpine/releases/tag/v3.10.3)). I tested on prior versions, and I don't see the error.

I've created a demo repository with a fresh Laravel install with Turbo on it ([link](https://github.com/tonysm/alpine-turbo-transition-issue)). Updated the README with instructions on how to test the app and reproduce the issue.

**Comment by ekwoka** [comment]
I just has to do with some effects being queued to run before the old elements removal but then the element is removed and cleaned up before the effect runs.

This kind of thing can happen any time elements are removed if they have a lot of directives.

If you're looking st using Laravel, turbo, and Alpine, why not check out Laravel Livewire?

**Comment by calebporzio** [comment]
I pulled down the repo and was able to reproduce with my local Alpine version.

You're calling Alpine.start() twice which is causing two of every Alpine feature to register on each element and they are bumping into each other with the transition cancellation (trying to cancel a transition that was already cancelled)

I'm going to PR a feature to provide a console warning when Alpine is being started twice on the same page today. That should help with things like this. But for now, the fix for you is removing one of those Alpine.start()s:

<img width="1433" alt="image" src="https://github.com/alpinejs/alpine/assets/3670578/1b665a79-b132-41a0-bd47-505724a3ca32">

**Reply by tonysm** [reply]
Dude, I didn't notice I was calling it twice. 🤦🏼

**Comment by tonysm** [comment]
As @calebporzio pointed out, I was starting Alpine twice, which was triggering the original error. I've updated the repo to remove that duplication; that specific error doesn't happen anymore. I thought that was it, but then I tried to wrap the main nav LW component around a `[data-turbo-permanent]` element ([commit](https://github.com/tonysm/alpine-turbo-transition-issue/commit/bcd741479ae872d0f8f9420a2e00c5523d1a4b22)), and I still got a similar error.

<img width="652" alt="Screenshot 2023-05-10 at 12 00 52" src="https://github.com/alpinejs/alpine/assets/1178621/18242b28-983e-4e72-8ae8-294c94bf432f">

I tried reproducing it without a LW component, but it works when I wrap other things. It may have something to do with LW. And it behaves similarly to the other one: no errors on 3.10.2, fails on 3.10.3.
